src/core/server/http/http_service.ts

```diff
@@ -86,14 +97,46 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
         await this.httpsRedirectServer.start(config);
       }
 
-      await this.httpServer.start(config);
+      await this.httpServer.start();
+      await Promise.all([...this.secondaryServers.values()].map(server => server.start()));
     }
 
     return {
-      isListening: () => this.httpServer.isListening(),
+      isListening: (port = 0) => {
+        const server = this.secondaryServers.get(port);
+        if (server) return server.isListening();
```

Comment:
should we differentiate between primary server and secondary? that creates additional logic

Reply:
i originally didn't differentiate them but thought we might want to since we might want ot know what the primary instance is (than again we could find the port for primary and get it that way).  

anyway i created an issue for it https://github.com/elastic/kibana/issues/37695.